### PR TITLE
GH-1802 Refactor Persist.discover

### PR DESCRIPTION
**Pull request by GordonSmith**
Move into PropertyExt

Fixes GH-1802

Signed-off-by: Gordon Smith <gordonjsmith@gmail.com>

### CODE REVIEW

**Comment by GordonSmith** [comment]
@jbrundage Please Review

File: src/common/PropertyExt.js [hunk 1]
```diff
@@ -154,16 +154,36 @@
...
-                retVal.push(this[key]);
+                var meta = this[key];
+                if (expandProxies && meta.type) {
+                    var item = this;
+                    while (meta.type === "proxy") {
+                        item = item[meta.proxy];
+                        meta = item.publishedProperty(meta.method);
+                    }
+                    if (meta.id !== this[key].id) {
+                        meta = JSON.parse(JSON.stringify(meta));  //  Clone meta so we can safely replace the id.
```
Comment (jbrundage) [inline src/common/PropertyExt.js]: I think this will cause problems with "function" and potentially "array"/"object" publish parameters.

Comment (GordonSmith) [inline src/common/PropertyExt.js]: @jbrundage:
1.  This is the same code that is currently in use, just refactored from Persist.js to PropertyExt.js
2. We don't don't serialize function properties that way (and our Widget Arrays shouldn't have circular references).

There probably is a more efficient way to do this, but I would rather leave the existing behavior in place and if we do change it, do it in a diff pull request?

**Comment by mlzummo** [comment]
looks good

**Pull request merged**